ursine/header_parsing.py

```diff
@@ -0,0 +1,58 @@
+'''Parsing for SIP URIs.'''
+from .uri import URI
+from collections import namedtuple
+
+
+HeaderParseResult = namedtuple('HeaderParseResult', (
+    'display_name',
+    'parameters',
+    'uri',
+))
+
+
+def parse_params(params_str):
+    params = {}
+    for pair in params_str.split(';'):
+        if pair == '':
+            continue
+        key, eq, val = pair.partition('=')
+        if len(key) == 0 or eq != '=':
+            raise ValueError(f'invalid uri parameter `{pair}`')
+        params[key] = val
+    return params
+
+
+def parse_display_name(dsp):
+    if dsp is None:
+        return None
+    if '"' in dsp:
+        quoted_parts = dsp.split('"')
```

Comment:
If you're looking for 3 splits, you can specify the maximum number of splits to perform: `dsp.split('"', 4)`. If more than 3 is too many, might as well stop at 4. (Imagine if I give you malformed data of two hundred `"` chracters :wink:)